aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -134,15 +134,28 @@ private HandlerMethod<ResourceModel, CallbackContext> safeAddTags(
             final HandlerMethod<ResourceModel, CallbackContext> handlerMethod,
             final Tagging.TagSet allTags
     ) {
-        return (proxy, rdsProxyClient, progress, tagSet) -> {
-            final ProgressEvent<ResourceModel, CallbackContext> result = handlerMethod.invoke(proxy, rdsProxyClient, progress, allTags);
-            if (HandlerErrorCode.AccessDenied.equals(result.getErrorCode())) {
-                final Tagging.TagSet systemTags = Tagging.TagSet.builder().systemTags(allTags.getSystemTags()).build();
-                return handlerMethod.invoke(proxy, rdsProxyClient, progress, systemTags);
+        return (proxy, rdsProxyClient, progress, tagSet) -> progress.then(p -> {
+            final CallbackContext context = p.getCallbackContext();
+            if (context.isSoftFailTags()) {
+                return p;
+            }
+            final ProgressEvent<ResourceModel, CallbackContext> result = handlerMethod.invoke(proxy, rdsProxyClient, p, allTags);
```

Comment:
nit: Should we name this result something more meaningful?

Reply:
I believe the scope of this lambda is very compact, hence the naming